arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Iterable.kt

```diff
@@ -1287,6 +1317,21 @@ public fun <A, K, V> Iterable<A>.crosswalkMap(f: (A) -> Map<K, V>): Map<K, List<
     }
   }
 
+/**
+ * Applies function [f] to each element
+ * and returns the result Iterable<B> without null.
+ *
+ * ```kotlin
+ * import arrow.core.crosswalk
+ * import io.kotest.matchers.shouldBe
+ *
+ * fun test() {
+ *   val ints = listOf(1, 2)
+ *   val res = ints.crosswalkNull { i -> if (i % 2 == 0) "x${i}" else null }
+ *   res shouldBe listOf("x2")
+ * }
+ * ```
+ */
 public fun <A, B> Iterable<A>.crosswalkNull(f: (A) -> B?): List<B>? =
```

Comment:
Passing `emptyList()` as the first argument to fold will ensure that `bs` is always non-null, so the results of `Ior.fromNullables` and `croswalkNull` will also be non-null. If you expect `Iterable<A>` to be empty or the result of applying the function `f` to it will all be null, it seems better that the first argument of fold is also null.

Reply:
@serras Thank you for your reply.

> Does this make sense?

I have the same understanding as you have written.

> Does this coincide with the behavior here?

However, the behavior seems not to coincides with the understandings.
The first and third case of following tests fails.

```kotlin
class CrosswalkNullTest : FunSpec({
    test("return null if f(it) returns null") {
        val list = listOf(1, 2, 3)
        val result = list.crosswalkNull { null }
        // This test fails because result is emptyList()
        result shouldBe null
    }

    test("return emptyList() if Iterable is empty") {
        val list = emptyList<Int>()
        val result = list.crosswalkNull { it.toString() }
        // This test passes
        result shouldBe emptyList()
    }

    test("return non-empty if Iterable is non-empty and f(it) returns not-null") {
        val list = listOf(1, 2, 3)
        val result = list.crosswalkNull { it.toString() }
        // This test fails because result is listOf("3", "2", "1")
        result shouldBe listOf("1", "2", "3")
    }
})
```